packages/block-library/src/site-logo/editor.scss

```diff
@@ -108,6 +108,12 @@
 	}
 }
 
+// Reduced specificity for the placeholder border radius,
+// so that border-radius from global styles can be applied.
+:root :where(.wp-block-site-logo .components-placeholder) {
+	border-radius: inherit;
+}
+
```

Comment:
Even with the `border-radius:inherit `
1) Commented out on line 58
2) Moved from line 67 to 114 with reduced specificity,
the placeholder component own CSS still overrides the border radius from the global styles.